components/SiteNavigation.vue

```diff
@@ -32,101 +32,6 @@
-          {
-            title: 'Requesting Payment',
-            path: '/guides/requesting-payment',
-          },
-          {
-            title: 'QR Code Flow',
-            path: '/guides/merchant-integration-qr-code-flow',
-          },
-          {
-            title: 'Barcode Flow',
-            path: '/guides/merchant-integration-barcode-flow',
-          },
-          {
-            title: 'Error Handling',
-            path: '/guides/merchant-integration-error-handling',
-          },
-          {
-            title: 'Transaction Reporting',
-            path: '/guides/transaction-reporting',
-          },
-          {
-            title: 'Initiating Refunds',
-            path: '/guides/initiating-refunds',
-          },
-          {
-            title: 'Line Items',
-            path: '/guides/line-items',
-          },
-          {
-            title: 'Payment Conditions',
-            path: '/guides/payment-conditions',
-          },
-          {
-            title: 'Requesting Pre Auth',
-            path: '/guides/requesting-pre-auth',
-          },
-          {
-            title: 'Patron Not Present',
-            path: '/guides/patron-not-present',
-          },
-        ]
-      }
-    ]
-  },
-  {
-    title: 'Connections',
-    path: '/connections',
-    icon: 'connections',
-    children: [
-      {
-        title: 'Farmlands',
-        path: '/connections/farmlands',
-        children: [
-          {
-            title: 'POS Integration Guide',
-            path: '/guides/farmlands-pos-integration',
-          }
-        ]
-      }
-    ]
-  },
-  {
-    title: 'API',
-    path: config.baseUrl + '/api',
-    icon: 'settings',
-  }
-];
-
-const urlToActiveNav = {
-  '/guides/farmlands-pos-integration': '/connections/farmlands/farmlands-pos-integration',
-  '/guides/payment-conditions': '/reference/merchant-integrations/payment-conditions',
-  '/guides/requesting-pre-auth': '/reference/merchant-integrations/requesting-pre-auth',
-  '/guides/patron-not-present': '/reference/merchant-integrations/patron-not-present',
-  '/guides/merchant-integration-error-handling': '/reference/merchant-integrations/merchant-integration-error-handling',
-  '/guides/line-items': '/reference/merchant-integrations/line-items',
-  '/guides/initiating-refunds': '/reference/merchant-integrations/initiating-refunds',
-  '/guides/point-of-sale': '/reference/merchant-integrations/point-of-sale',
-  '/guides/merchant-integration-qr-code-flow': '/reference/merchant-integrations/merchant-integration-qr-code-flow',
-  '/guides/transaction-reporting': '/reference/merchant-integrations/transaction-reporting',
-  '/guides/requesting-payment': '/reference/merchant-integrations/requesting-payment',
-  '/guides/merchant-integration-barcode-flow': '/reference/merchant-integrations/merchant-integration-barcode-flow',
-};
+const content = await queryContent().find();
+const { menu, pathToActiveNav } = navigation.create({ baseUrl: config.baseUrl, content });
```

Comment:
It looks like `pathToActiveNav ` is not inside this component directly. I'm wondering if we should decouple building the data structures from retrieving them so that different components can pull in what they need?
-  `SiteNavigation.vue` uses the `menu data structure
- `SiteNavigationDisclosure.vue` uses the `pathToActiveNav` data structure

Reply:
Yup, I've created separate getters for the `menu` data structure and the `pathToActiveNav` data structure so the components can just get what they need.